tests/integration/test_duckdb_tpch.py

```diff
@@ -0,0 +1,108 @@
+import duckdb
+import pytest
+
+from ..common import SubstraitUtils
+from ..consumers.duckdb_consumer import DuckDBConsumer
+from ..parametrization import custom_parametrization
+from ..verification import verify_equals
+from .queries.tpch_test_cases import TPCH_QUERY_TESTS
+
+
+@pytest.mark.usefixtures("prepare_tpch_parquet_data")
+class TestDuckDBConsumer:
+    """
+    Test Class for testing Substrait using Acero as a consumer.
+    """
+
+    @staticmethod
+    @pytest.fixture(scope="class", autouse=True)
+    def setup_teardown_class(request):
+        cls = request.cls
+
+        cls.db_connection = duckdb.connect()
+        cls.db_connection.execute("INSTALL substrait")
+        cls.db_connection.execute("LOAD substrait")
+        cls.consumer = DuckDBConsumer()
+        cls.utils = SubstraitUtils()
+        cls.created_tables = set()
+
+        yield
+
+        cls.db_connection.close()
+
+    @custom_parametrization(TPCH_QUERY_TESTS)
+    def test_substrait_query(
+        self,
+        test_name: str,
+        file_names: list,
+        sql_query: str,
+        substrait_query: str,
+        sort_results: bool = False,
+    ) -> None:
+        """
+        1.  Load all the parquet files into DuckDB as separate tables.
+        2.  Format the SQL query to work with DuckDB by inserting all the table names.
+        3.  Execute the SQL on DuckDB.
+        4.  Run the substrait query plan.
+        5.  Compare substrait query plan results against the results of
+            running the SQL on DuckDB.
+
+        Parameters:
+            test_name:
+                Name of test.
+            file_names:
+                List of parquet files.
+            sql_query:
+                SQL query.
+            sort_results:
+                Whether to sort the results before comparison.
+        """
+
+        parquet_file_paths = self.utils.get_full_path(file_names)
+
+        # Load the parquet files into DuckDB and return all the table names as a list
+        table_names = self.consumer.load_tables_from_parquet(
+            self.db_connection, self.created_tables, file_names, parquet_file_paths
+        )
+
+        # Format the sql query by inserting all the table names
+        sql_query = sql_query.format(*table_names)
+
+        # Convert the SQL into a substrait query plan and run the plan.
+        subtrait_query_result_tb = self.consumer.run_substrait_query(
+            self.db_connection, sql_query
+        )
```

Comment:
It seems this function is DuckDB consuming its own produced substrait plans, right?

If that's the case (and goal), you can simplify the code by setting `PRAGMA enable_verification`.

Basically, when that pragma is set, DuckDB will internally execute the query and the substrait plan resulting from `get_substrait(SQL)` and compare it's resulting throwing an Error if they do not match.

I think the verify_equals is more interesting if you are comparing results of two different systems consuming a substrait plan.

Reply:
> It seems this function is DuckDB consuming its own produced substrait plans, right?
> 
> If that's the case (and goal), you can simplify the code by setting `PRAGMA enable_verification`.
> 
> Basically, when that pragma is set, DuckDB will internally execute the query and the substrait plan resulting from `get_substrait(SQL)` and compare it's resulting throwing an Error if they do not match.
> 
> I think the verify_equals is more interesting if you are comparing results of two different systems consuming a substrait plan.

Yes, that's correct.  However, the eventual plan is to have tests where DuckDB is consuming substrait plans produced by other producers as well.